### ISSUE TITLE
Incorrect mapping field 

### ISSUE THREAD

**Issue by Ormanko** [vapor/fluent#497]
I have a models

```swift
final class Product: Codable {
    var id: Int?
    var productname: String
    var partnumber: String
    var measurementID: Measurement.ID
    var categoryID: Category.ID
   
    init(productname: String, partnumber:String, measurement: Measurement.ID, category: Category.ID) throws {
        self.partnumber = partnumber
        self.productname = productname
        self.measurementID = measurement
        self.categoryID = category
    }
}

extension Product: Migration {
    static func prepare(on connection: SQLiteConnection) -> Future<Void> {
        return Database.create(self, on: connection) { builder in
            try addProperties(to: builder)
            try builder.addReference(from: \.measurementID, to: \Measurement.id)        
        }
    }   
}

extension Product {
    var measurement: Parent<Product, Measurement> {
        return  parent(\.measurementID)
    }
}
final class Measurement: Codable {
    var id: Int?
    var measurementname: String
   
    init(measurementname: String) {
        self.measurementname = measurementname
    }
}
extension Measurement {
    var products: Children<Measurement, Product> {
        return children(\.measurementID)
    }
}
```

And view handler

```swift
    struct IndexTable: Encodable {
        let id1:Int
        let partnumber: String
        let productname: String
        let measurement: String
    }
    struct IndexContext: Encodable {
        let title: String
        let products: [IndexTable]?
    }

 func indexHandler(_ req: Request) throws -> Future<View> {
        return try Product.query(on: req).join(Measurement.self, field: \Measurement.id, to: \Product.measurementID).alsoDecode(Measurement.self).all()
                .flatMap(to: View.self) { products in
                    var contextdata: [IndexTable] = []
                    products.forEach({(product, measurement) in
                        let cd = IndexTable(id1: product.id!, partnumber: product.partnumber, productname: product.productname, measurement: measurement.measurementname)
                    contextdata.append(cd)
            })
            let context = IndexContext(title: "Homepage", products: contextdata)
            return try req.view().render("index", context)
        }
    }

```
This mapping is not correct
```swift
let cd = IndexTable(id1: product.id!, partnumber: product.partnumber, productname: product.productname, measurement: measurement.measurementname)
```
product.id! when mapping is measurement.id

May be I doing is incorrect? maybe there is an options to organize a index select with join (or without join) more beautifully?

**Comment by tanner0101** [comment]
That seems like a bug. `alsoDecode(...)` should work like you have it. Which database are you using?

**Comment by Ormanko** [comment]
Sorry. SQLite. I am not testing with other databases.

**Comment by tanner0101** [comment]
@Ormanko this was fixed in `SQLite 3.0.0 RC 3`. Thanks for reporting!
